src/lib_json/json_value.cpp

```diff
@@ -104,7 +111,11 @@ static inline char* duplicateStringValue(const char* value,
  */
 static inline char* duplicateAndPrefixStringValue(
```

Comment:
Hi Chris, thanks again, sorry my brain is having problems working through all the use cases, so the reasoning step by step:

1. I had to have a macro in the method prototypes because i cannot add a value anywhere so that in the case of non-secure mode i have nothing to pass in there (couldn't add a static variable because this would change the binary compatibility, could add locally but allocating and deallocating this seemed expensive when unecessary).
1. In Value it's clear, the copy or create methods will prefix the string so no need here to track the length separately
1. CZString does not use the prefixing strategy but you can use storage_.length_ to get the amount of memory that was allocated
1. CommentInfo does not store the length allocated anywhere to be properly zeroed

So what i'll do for the moment is to then use the lengths where i have it available and guess then for CommentInfo i think; would you think this ok?